### PR TITLE
Pivotal ID # 187456681: Submission request file status

### PR DESCRIPTION
**Pull request by Juan-EBI**
https://www.pivotaltracker.com/story/show/187456681
- added request cleaner indexer

### CODE REVIEW

File: submission/submission-core/src/test/kotlin/ac/uk/ebi/biostd/submission/domain/request/SubmissionRequestCleanIndexerTest.kt [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+            mockkStatic(ExtSerializationService::filesFlow)
+            coEvery { serializationService.filesFlow(currentSub) } returns flowOf(file)
+
+            testInstance.indexRequest(newSub)
+
+            coVerify(exactly = 0) { fileRqtService.saveSubmissionRequestFile(any()) }
+        }
+
+        @Test
+        fun `when a file of the current submission need to be replaced`(
```
Comment (jhoanmanuelms) [inline submission/submission-core/src/test/kotlin/ac/uk/ebi/biostd/submission/domain/request/SubmissionRequestCleanIndexerTest.kt:138]: typo: needs to be replaced

File: submission/submission-core/src/test/kotlin/ac/uk/ebi/biostd/submission/domain/request/SubmissionRequestCleanIndexerTest.kt [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+    @Test
+    fun `when no current submission`() =
+        runTest {
+            every { newSub.accNo } returns ACC_NO
+            every { newSub.version } returns CURRENT_VERSION
+            coEvery { queryService.findExtByAccNo(ACC_NO, includeFileListFiles = true) } returns null
+
+            val files = testInstance.indexRequest(newSub)
+
+            assertThat(files).isZero()
```
Comment (jhoanmanuelms) [inline submission/submission-core/src/test/kotlin/ac/uk/ebi/biostd/submission/domain/request/SubmissionRequestCleanIndexerTest.kt:55]: we should assert no calls to the serialization service

File: submission/submission-core/src/test/kotlin/ac/uk/ebi/biostd/submission/domain/request/SubmissionRequestCleanIndexerTest.kt [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+
+            every { currentSub.storageMode } returns STORAGE_MODE
+            every { currentSub.accNo } returns ACC_NO
+            every { currentSub.version } returns CURRENT_VERSION
+
+            coEvery { queryService.findExtByAccNo(ACC_NO, includeFileListFiles = true) } returns currentSub
+        }
+
+        @Test
+        fun `when a files has the same md5 but diferent path so file need to be cleaned`(
```
Comment (jhoanmanuelms) [inline submission/submission-core/src/test/kotlin/ac/uk/ebi/biostd/submission/domain/request/SubmissionRequestCleanIndexerTest.kt:76]: typo: ~~files~~ file

File: submission/submission-core/src/main/kotlin/ac/uk/ebi/biostd/submission/domain/request/SubmissionRequestCleanIndexer.kt [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+        val newFile = files[existing.filePath]
+        return when {
+            newFile == null -> DELETE_AFTER_SUBISSION
+            newFile.md5 != existing.md5 && storageMode == existing.storageMode -> DELETE_NOW
+            else -> MatchType.RE_USED_FILE
+        }
+    }
+}
+
+private enum class MatchType {
```
[the remark this answers is not in the document]
Comment (awaisathar) [reply submission/submission-core/src/main/kotlin/ac/uk/ebi/biostd/submission/domain/request/SubmissionRequestCleanIndexer.kt:96]: REUSED would work better :)

File: submission/persistence-common-api/src/main/kotlin/ac/uk/ebi/biostd/persistence/common/model/SubmissionRequestFile.kt [hunk 5]
```diff
@@ -24,4 +24,6 @@ enum class RequestFileStatus {
     LOADED,
     COPIED,
     RELEASED,
+    TO_CLEAN_NOW,
```
[the remark this answers is not in the document]
Comment (Juan-EBI) [reply submission/persistence-common-api/src/main/kotlin/ac/uk/ebi/biostd/persistence/common/model/SubmissionRequestFile.kt:27]: I like them, fantastic suggestions

**Pull request merged**